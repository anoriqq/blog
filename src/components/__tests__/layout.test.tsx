import React from 'react';
import renderer from 'react-test-renderer';
import { PageRendererProps } from 'gatsby';

import Layout from '../layout';

describe('Layout', () => {
  it('renders correctly of location at "/"', () => {
    const location = { pathname: '' } as PageRendererProps['location'];
    const tree = renderer
      .create(
        <Layout location={location} title="All posts">
          <div>Hello World</div>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly of location at "hello-world"', () => {
    const location = {
      pathname: 'hello-world',
    } as PageRendererProps['location'];
    const tree = renderer
      .create(
        <Layout location={location} title="All posts">
          <div>Hello World</div>
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
