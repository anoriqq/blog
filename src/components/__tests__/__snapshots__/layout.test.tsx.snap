// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly of location at "/" 1`] = `
<div
  style={
    Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "42rem",
      "padding": "2.625rem 1.3125rem",
    }
  }
>
  <header>
    <h3
      style={
        Object {
          "fontFamily": "Montserrat, sans-serif",
          "marginTop": 0,
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "boxShadow": "none",
            "color": "inherit",
          }
        }
      >
        All posts
      </a>
    </h3>
  </header>
  <main>
    <div>
      Hello World
    </div>
  </main>
  <footer>
    © 2020, Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;

exports[`Layout renders correctly of location at "hello-world" 1`] = `
<div
  style={
    Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "42rem",
      "padding": "2.625rem 1.3125rem",
    }
  }
>
  <header>
    <h3
      style={
        Object {
          "fontFamily": "Montserrat, sans-serif",
          "marginTop": 0,
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "boxShadow": "none",
            "color": "inherit",
          }
        }
      >
        All posts
      </a>
    </h3>
  </header>
  <main>
    <div>
      Hello World
    </div>
  </main>
  <footer>
    © 2020, Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;
